### DISCUSSION TITLE
How to integrate web3Model via viem?

### DISCUSSION THREAD

**Discussion by denniarems** [WalletConnect/walletconnect-monorepo#2518] [Developer Support] [answered]
How to integrate web3model by viem provider setup. 
[Viem Custom Provider ](https://viem.sh/docs/clients/transports/custom.html)

i'm developing on vue js with typescript. 

the reason for making this i want integrate [Web3Auth]( https://web3auth.io/docs/connect-blockchain/polygon) also 

so how to solve this issue ?

**Comment by glitch-txs** [comment, marked as answer]
You can use ethereum-provider with viem as custom provider

https://docs.walletconnect.com/2.0/web/providers/ethereum
